### ISSUE TITLE
Crunz sends me an empty error email on every job

### ISSUE THREAD

**Issue by kinoute** [lavary/crunz#177]
**Crunz version**: 1.11.0-rc.1

**PHP version**: 7.3.0

**Operating system type and version**: Ubuntu 16.04.5

**Description**  
Despite generating no error, Crunz emails me on every task ("reporting error for event..."). With  log_errors: true, we can confirm there is no error shown:

`[2018-12-26 17:45:07] crunz.ERROR: Results(cd xx; php xx.php)    [] []`

**How to reproduce**  
Turn the email_errors to true on the latest version, add your email credentials, create some dummy task. Run it, it should email you with nothing.

**Possible Solution**  
Haven't one sorry.

**Comment by PabloKowalczyk** [comment]
Interesting, that was fixed in https://github.com/lavary/crunz/issues/161. Is there a regression? Does it work on `v1.11.0-beta.2`?

**Comment by kinoute** [comment]
Actually, when I reported #161, it was different. In #161, I wasn't getting an email on every task. I was getting correctly emails when my script generated one, but the email was empty (no description of the error).

Here, all my crunz tasks generate an error email, no matter if there is an error or not.

**Comment by kinoute** [comment]
Since it logs every task as an error now in the error log file, I monitor it. I see on some of them a ` Segmentation fault (core dumped)` message, like:

`[2018-12-26 18:40:07] crunz.ERROR: New (cd new; php new.php)  Segmentation fault (core dumped)   [] []`

Don't know if that can help you... I updated to PHP 7.3.0 by the way.

**Comment by kinoute** [comment]
I'm investigating the `Segmentation fault (core dumped)` issue, it could be related to mysql according to gdb. But even on tasks where there is no segfault, crunz still logs as if there was an error but it shows nothing.

Actually I was wrong, it doesn't say there is an (empty) error on every task, just a few. Every time they are launched. I removed the cronjob temporary, and run one of them manually with` schedule:run -vvv` to see if I could get more infos but it just says:

> Task New status: fail.
> Invoke Schedule's ping after.
> There is no ping after url.

The script `schedule:run` doesn't stop afterwards, it keeps going. When I do schedule:run manually on other time to check other jobs (that don't give "errors"), it stops after properly.

**Comment by PabloKowalczyk** [comment]
Maybe there are more info in PHP logs? What about your `display_error` setting?

> The script schedule:run doesn't stop afterwards, it keeps going.

You mean, Crunz should stop on failed task?

**Comment by kinoute** [comment]
I don't know what crunz should do exactly (or what it does in this case), but I thought that the `schedule:run` process not stopping after a failed task might tell you something about my problem.

On my crunz-project dumb project I made for for the other issue ("No task found"), if I make sure my script is going to throw an error, when I run `schedule:run -vvv`, it ends properly (I can type again in the terminal).

On my production server, if I run `schedule:run -vvv` manually, the task "fails" according to crunz (when it's not), and the `schedule:run` script is still running, like if it was waiting/doing something (I can't type on the terminal). I need to stop it manually with CTRL+C.

**Comment by PabloKowalczyk** [comment]
Crunz runs all tasks in parallel and IMO there is no need to stop all tasks because one task fail. Actually it is better that all other tasks still can be executed even if one failed. So in production Crunz run all tasks and wait for the results.
If you want you can use something like `schedule:run --task 12` to run only specific task, numbers can be taken from output of `schedule:list`.

If you think that errors could be reported better with `-vvv` please open new issue, we will try to improve it :)

**Comment by kinoute** [comment]
It seems to be related to MySQL. First the segfault error core dump indicated that, secondly, even when I don't get a segfault, crunz reports an empty error when I play with MySQL (simple query/bind/execute stuff). It Might be related to the milli/macroseconds change in PHP 7.3.0 when dealing with datetime(6) or timestamp(6) column I'm not sure. I reported the bug here: https://bugs.php.net/bug.php?id=77358

Still wondering why and where crunz decides there is an error, what triggers it. I tried to `print_r `some stuff in the crunz library while it "fails" but I didn't get anything interesting. Still blank.

**Comment by PabloKowalczyk** [comment]
Crunz simply check if process was successful - https://github.com/lavary/crunz/blob/master/src/EventRunner.php#L168, under the hood is `symfony/process` lib, which, also, simply check is process return code was different than 0 - https://github.com/symfony/process/blob/master/Process.php#L768.
There is no magic, in Crunz or Symfony, your scripts simply ends with return code different than 0.

**Comment by kinoute** [comment]
Yeah, apparently I'm getting "1" as a return code every time I bind parameters (even if the script goes well and does the job). I will investigate. Sorry for the inconvenience

**Comment by PabloKowalczyk** [comment]
No problem :)
